core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -91,181 +93,224 @@ public static void processParquetPartition(
-                        oooColumns,
-                        sortedTimestampsAddr,
-                        tableWriter,
-                        partitionDecoder,
-                        rowGroupBuffers,
-                        rowGroup - 1,
-                        timestampIndex,
-                        timestampColumnType,
-                        mergeRangeLo,
-                        mergeRangeHi,
-                        tableWriterMetadata,
-                        srcOooBatchRowSize
-                );
-                mergeRangeLo = mergeRangeHi + 1;
+                if (mergeRangeLo <= srcOooHi) {
+                    // merge the tail [mergeRangeLo, srcOooHi] into the last row group
+                    // this also handles the case where there is only a single row group
+                    mergeRowGroup(
+                            partitionDescriptor,
+                            partitionUpdater,
+                            columnIdsAndTypes,
+                            oooColumns,
+                            sortedTimestampsAddr,
+                            tableWriter,
+                            partitionDecoder,
+                            rowGroupBuffers,
+                            rowGroupCount - 1,
+                            timestampIndex,
+                            timestampColumnType,
+                            mergeRangeLo,
+                            srcOooHi,
+                            tableWriterMetadata,
+                            srcOooBatchRowSize
+                    );
+                }
+            } finally {
+                ff.munmap(parquetAddr, parquetSize, MemoryTag.MMAP_PARQUET_PARTITION_DECODER);
             }
 
-            if (mergeRangeLo <= srcOooHi) {
-                // merge the tail [mergeRangeLo, srcOooHi] into the last row group
-                // this also handles the case where there is only a single row group
-                mergeRowGroup(
-                        partitionDescriptor,
-                        partitionUpdater,
-                        columnIdsAndTypes,
-                        oooColumns,
-                        sortedTimestampsAddr,
-                        tableWriter,
-                        partitionDecoder,
-                        rowGroupBuffers,
-                        rowGroupCount - 1,
-                        timestampIndex,
-                        timestampColumnType,
-                        mergeRangeLo,
-                        srcOooHi,
-                        tableWriterMetadata,
-                        srcOooBatchRowSize
-                );
-            }
-        } catch (Throwable e) {
+            // Update indexes
+            updateParquetIndexes(
+                    partitionBy,
+                    partitionTimestamp,
+                    tableWriter,
+                    srcNameTxn,
+                    o3Basket,
+                    newPartitionSize,
+                    pathToTable,
+                    path,
```

Comment:
I don't like the idea of using `getThreadLocal`/`getThreadLocal2` since there is always a chance that the calling code uses one of them already and, in the end, it's hard to debug such issues. To me, resetting the path a few times isn't a big deal. WDYT?